### DISCUSSION TITLE
Development opf e107 v2.3.3 via xampp on a usb drive

### DISCUSSION THREAD

**Discussion by NECPWAweb** [e107inc/e107#5196] [Q&A Installation] [answered]
As a long term user of e107 but still on v1.0.5 (I know but it is modified a bit and running a responsive theme here https://necpwa.org/ ) I want to test upgrading on a dev site before committing to an upgrade so installed xampp portable on a USB drive but when I go to localhost/install to run v2.3.3 I get: 

Error
A minimum version of PHP 5.6 is required

Can anyone throw any light on what the problem is please as php is supposedly included?

![no php](https://github.com/e107inc/e107/assets/160168219/bafc02c9-3e7b-40c9-b31f-115a85ef7fa7)
 
I did try on my desktop but running win 11 insider builds xampp doesn't want to play at all because of security issues and errors.

**Comment by tgtje** [comment]
@NECPWAweb if you are using the 'official' usb packs (vc9) likely highest in php is 5.4. ( a lot of php changed since that; 2014)
The old pieces should run, however 2.3.3 does not support it anymore.
Version 1 of e107 has gone to v2 with newer coding, and a lot of changes.
Due to the demanding higher php versions (security driven by hosting also) a higher php version will disrupt a v1 installment (non functional pieces andd others demand php 5.6 and up where at least 7.4 is already to be seen as a new minimum.

Long story short if you still like to use usb driven. Try [usbwebserver](https://usbwebserver.yura.mk.ua/).
However take a 5.6 version first : create the old one on this. : after that : update this with a v2.0.0 version ([source forge](https://sourceforge.net/projects/e107/files/e107/)) so it can run on a higher php version. (NOTE : items like deprecated plugins etc. can brake all) hence all is on v2 and up these days.

From 5.6 onwards you could try to do things although i would suggest to end on 7.4 at least.
So manual labour : from low grade up (php +system); grade up again etc.. helas no real other solution.

You could try as mentioned usbwebserver on 7.4 or so, but no guarantee your v1 version will not criple... (outcome would be a good impression what you can expect to do/and errors).
Of course alternative you could occupy a new install with database table contents, but that (depending on used stuff) is a tricky time consuming thing.

**Reply by NECPWAweb** [reply]
Thanks for the prompt reply. My sites currently run php 5.6 and I mistakenly thought that the xampp version I installed ran php 5.6, (or higher) now I know better and will try usbserver. I will post the results when I run through it all for others in case they have similar issues but it may be a while.

**Reply by NECPWAweb** [reply]
#You could try as mentioned usbwebserver on 7.4 or so, but no guarantee your v1 version will not criple... (outcome would be a good impression what you can expect to do/and errors).

usbwebserver did let me get started, but to be honest, it's slow in the admin area..... running on a USB stick. Anyway, I have spent the day trying it out and eventually installed the php 7.4 version of usbwebserver and installed e107 v2.2.1 - yes a couple of old plugins won't work, but that can be worked around. 

However, the theme I currently run didn't appear compatible as all I got was a blank page so looked like I was going to have to create a new one before doing anything about upgrades on the production sites, 

BTW I did try the update tool in theme manager but couldn't get the 'keywords' section to accept anything I entered so the tool wouldn't run and create the xml file. However, I took the xml file from the 'blank' theme, edited that and put it in my theme folder and et voila it now appears to work!!

**Reply by Alex-e107nl** [reply, marked as answer]
Hi you can use the uniformserver, i have made a working server for your usb stick, it's optimized and as fast as you would expect :-)

[https://e107.nl/e107_plugins/download/download.php?view.5](https://e107.nl/e107_plugins/download/download.php?view.5)

You can install multiple versions of e107, but now you have at least a working example with version 2.3.3..

**Reply by NECPWAweb** [reply]
Thanks, that was very easy to install and I can now play around  with things to see what I need to do, upgrade or start again with a clean install and recreate the pages.

**Reply by Alex-e107nl** [reply]
![userver_main](https://github.com/e107inc/e107/assets/11962513/6755fff7-6471-4841-8c4b-86b83c8acd9a)
?-)

**Comment by NECPWAweb** [comment]
My Bad, I eventually did simply press the button.....

Thanks for taking the time to build userver and e107 v2.3.3 and giving me access, it's much appreciated and makes life so much easier :)

**Comment by Alex-e107nl** [comment]
NP, but when you install another version of e107 or if you do an update please keep in mind that because it is a windows server the .htaccess is not standard e107...  When you read the info on [https://www.uniformserver.com/](https://www.uniformserver.com/) you can also switch between php versions...

**Reply by NECPWAweb** [reply]
Thanks. Managed to find and install php7 but can't find a plugin for php5. Shame as I could have started with an e107 v1.5 install and run through the complete upgrade to v2.3 locally with a copy of my existing database.

**Reply by tgtje** [reply]
try [here](https://sourceforge.net/projects/miniserver/files/Old%20Files/UniServer%20Zero/) although old and working ? php 5.4 versions : read info in folders 

and e107 v1.5  ??  highest is 1.0.4 as latest of the v1 range

**Reply by NECPWAweb** [reply]
[e107 v1.5 ?? highest is 1.0.4 as latest of the v1 range] sorry it's v1.0.5 svn, slightly modified (templates, language files basically)

**Reply by NECPWAweb** [reply]
Thanks for the input and suggestions guys, but I have had a good play around with V2.3.3, and I really don't like parts of the backend. The 'pages' section is clumsy compared to the original (I have a lot of those on one site) and the 'navigation' is another area that has got more complex and not easy to get to grips with compared to the original 'site links'. 

I also couldn't get the admin theme to change to anything else for some reason so couldn't tell if it was simply that causing my issues, but for now I think I'll stick with V1 :(

**Reply by Alex-e107nl** [reply]
Yeah i know a lot has changed and not always in the right direction, but there's a lot possible, but you cannot stay on php 5.6 forever, i have seen your website and i was surprised that it was made with e107!

**Reply by NECPWAweb** [reply]
OK gave up using usb and uniformservers and now using a live website necpwa.website to test out the upgrade once I transfer the live database over. Via usb / uniform I have database errors going up via v2.0 to v 2.9 in php 5.6 so intend to try and go from v1.1.4 to v 2.3.3 direct with php 8.1

Thanks for the input and I'll post back the results (once I find time to do the upgrade)

**Reply by NECPWAweb** [reply]
Well the upgrade wasn't straightforward but it was 90% OK, Eventually sorted out the navigation and got the 'Forum' mostly working (had to change theme) but still trying to work out how to upload images as forum attachments as ATM that doesn't want to work.

Busy working through the custom page layouts as even uninstalling tinymce left the pages a mess so it's taking a bit of time. So the test site whilst still a work in progress is coming along https://necpwa.website/ (feedback appreciated)

**Reply by tgtje** [reply]
Just a quick reply site already looks good. As for downloads page bottom star (huge) follow here the postings, so you do not have to re-invent [link address](https://app.gitter.im/#/room/#e107inc_e107:gitter.im/$OPIywuLfdiTTAcqO41Xv7kgkp-IwTl2a4bbzQ-426R0:~:text=Voorbeeld%20van%20bericht-,1609817,-1)

**Reply by NECPWAweb** [reply]
Thanks eventually found the 'star' it in download template (which I already edited to remove the 'rating' column as I couldn't find anywhere in admin to turn ratings off

**Comment by NECPWAweb** [comment]
Eventually upgraded 2 sites from V1.1.4 to V2.3.3 via the test site and transferred both to the live sites and 1 worked fine, the second kept throwing a file not found error on downloads. It took a while (and a lot of heartache) before I eventually tracked down the reason.

The first site update the main .htaccess with the e107.htaccess code. The second didn't update the main .htaccess, why.. Who knows, but I resolved it copying the 1st sites .htaccess over.

Anyway both sites are now running v2.3.3 in PHP 7.4 (did try PHP 8.1 but got a blank screen on both sites)
